src/lib/operators/join_sort_merge.cpp

```diff
@@ -1354,7 +1335,8 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
           }
         }
 
-        for (size_t row = 0; row < left_null_rows_matches.size(); row++) {
+        const auto left_null_rows_match_count = left_null_rows_matches.size();
+        for (size_t row = 0; row < left_null_rows_match_count; ++row) {
```

Comment:
auto row

Reply:
If I use auto I get the error `comparison of integers of different signs: 'int' and 'const unsigned long'`. Do you have a hint on how we could still use auto?

Reply:
`size_t row = size_t{0};`